src/lib/output/plugins/JavascriptIndexPlugin.ts

```diff
@@ -66,12 +66,18 @@ export class JavascriptIndexPlugin extends RendererComponent {
             rows.push(row);
         }
 
-        const fileName = Path.join(event.outputDirectory, 'assets', 'js', 'search.js');
-        const data =
-            `var typedoc = typedoc || {};
+        const jsFileName = Path.join(event.outputDirectory, 'assets', 'js', 'search.js');
+        const jsData =
+            `console.warn('[typedoc] search.js file is deprecated. Please, use search.json file in your custom theme instead.');
+            var typedoc = typedoc || {};
             typedoc.search = typedoc.search || {};
             typedoc.search.data = ${JSON.stringify({kinds: kinds, rows: rows})};`;
 
-        writeFile(fileName, data, false);
+        writeFile(jsFileName, jsData, false);
+
+        const jsonFileName = Path.join(event.outputDirectory, 'assets', 'js', 'search.json');
```

Comment:
Related to this, I'd really like to switch to serializing a pre-built index https://lunrjs.com/guides/index_prebuilding.html and sending that instead. There's no reason we can't do as much work for clients before rendering as possible... probably fits best in a different PR though.

Reply:
I think it is good idea, I have implemented this approach for typedoc-default-themes [here](https://github.com/TypeStrong/typedoc-default-themes/pull/101)